ocaml/typing/typecore.ml

```diff
@@ -8419,6 +8428,13 @@ let report_error ~loc env = function
         "@[Function arguments and returns must be representable.@]@ %a"
         (Layout.Violation.report_with_offender
            ~offender:(fun ppf -> Printtyp.type_expr ppf ty)) violation
+  | Invalid_label_for_src_pos arg_label ->
+      Location.errorf ~loc
+        "A position argument must not be %s."
+        (match arg_label with
+        | Nolabel -> "unlabelled"
+        | Optional _ -> "optional"
+        | Labelled _ | Position _ -> assert false )
```

Comment:
Code style question: I see other parts of the code that like to assert false. Should I instead use a polymorphic variant?

Reply:
No, I think this is fine as is.